### ISSUE TITLE
Sidekiq Batch - wrong number of arguments (given 0, expected 1)

### ISSUE THREAD

**Issue by pbstriker38** [mperham/sidekiq#5689]
Ruby version: 3.1.2
Rails version: 7.0.4
Sidekiq / Pro / Enterprise version(s): 6.5.8 / 5.5.6

We had a Redis outage last night and when it came back we got a bunch of jobs hitting an `ArgumentError` in sidekiq batch.

![image](https://user-images.githubusercontent.com/5264240/206515543-098b263a-ebe1-4501-810f-4394aea48b80.png)

```
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch.rb:31:in `redis'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch/middleware.rb:156:in `add_failure'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch/middleware.rb:53:in `block in call'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch/middleware.rb:52:in `handle_interrupt'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch/middleware.rb:52:in `rescue in call'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-5.5.6/lib/sidekiq/batch/middleware.rb:30:in `call'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/middleware/chain.rb:179:in `block in invoke'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/middleware/chain.rb:182:in `invoke'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/processor.rb:169:in `block in process'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/processor.rb:136:in `block (6 levels) in dispatch'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/job_retry.rb:113:in `local'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/processor.rb:135:in `block (5 levels) in dispatch'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/rails.rb:14:in `block in call'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/reloader.rb:72:in `block in wrap'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/reloader.rb:71:in `wrap'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/rails.rb:13:in `call'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
/home/ruby/vendor/bundle/ruby/3.1.0/gems/sidekiq-6.5.8/lib/sidekiq/processor.rb:263:in `stats'
```

![image](https://user-images.githubusercontent.com/5264240/206516308-38e007dd-6f6b-4bcb-855f-0ae5c1023876.png)

**Comment by pbstriker38** [comment]
Should the jobs be retry-able after this updating to this version? I get this when I retry them.

![image](https://user-images.githubusercontent.com/5264240/206597500-34c90dbd-69d9-4220-b82a-d15c67fc8a13.png)

**Comment by mperham** [comment]
Oh fer crying out loud. Alright, 5.5.8 is now out with a test in the test suite.

**Comment by pbstriker38** [comment]
Thanks. I'll try it out tomorrow

**Comment by pbstriker38** [comment]
It worked 👍

**Comment by gmcabrita** [comment]
> Should the jobs be retry-able after updating to this version? I get this when I retry them.
> 
> ![image](https://user-images.githubusercontent.com/5264240/206597500-34c90dbd-69d9-4220-b82a-d15c67fc8a13.png)

@mperham seems like this is happening on Sidekiq Pro 7.0.4. Was the fix only for 5.5.8?

```
Sidekiq::Batch.redis(bid) { |c| c.hset(key, "pending", 0) }
                                       ^^^
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:154:in `block in add_failure'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/capsule.rb:85:in `block in redis'
	/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:65:in `block (2 levels) in with'
	/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `handle_interrupt'
	/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `block in with'
	/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `handle_interrupt'
	/app/vendor/bundle/ruby/3.1.0/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `with'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/capsule.rb:82:in `redis'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq.rb:73:in `redis'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch.rb:32:in `redis'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:154:in `add_failure'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:53:in `block in call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:52:in `handle_interrupt'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:52:in `rescue in call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-pro-7.0.4/lib/sidekiq/batch/middleware.rb:30:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/middleware/chain.rb:177:in `block in invoke'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-ent-7.0.2/lib/sidekiq-ent/unique.rb:154:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/middleware/chain.rb:177:in `block in invoke'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-ent-7.0.2/lib/sidekiq-ent/limiter/middleware.rb:39:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/middleware/chain.rb:177:in `block in invoke'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/metrics/tracking.rb:24:in `track'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/metrics/tracking.rb:120:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/middleware/chain.rb:177:in `block in invoke'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/middleware/chain.rb:180:in `invoke'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:173:in `block in process'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:140:in `block (6 levels) in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/job_retry.rb:113:in `local'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:139:in `block (5 levels) in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/rails.rb:15:in `block in call'
	/app/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
	/app/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/reloader.rb:72:in `block in wrap'
	/app/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
	/app/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.4/lib/active_support/reloader.rb:71:in `wrap'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/rails.rb:14:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:135:in `block (4 levels) in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:267:in `stats'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:130:in `block (3 levels) in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/job_logger.rb:13:in `call'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:129:in `block (2 levels) in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/job_retry.rb:80:in `global'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:128:in `block in dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/job_logger.rb:39:in `prepare'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:127:in `dispatch'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:172:in `process'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:82:in `process_one'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-instrumentation-sidekiq-0.22.0/lib/opentelemetry/instrumentation/sidekiq/patches/processor.rb:21:in `block in process_one'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-common-0.19.6/lib/opentelemetry/common/utilities.rb:87:in `block in untraced'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-api-1.1.0/lib/opentelemetry/trace.rb:72:in `block in with_span'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-api-1.1.0/lib/opentelemetry/context.rb:87:in `with_value'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-api-1.1.0/lib/opentelemetry/trace.rb:72:in `with_span'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-common-0.19.6/lib/opentelemetry/common/utilities.rb:87:in `untraced'
	/app/vendor/bundle/ruby/3.1.0/gems/opentelemetry-instrumentation-sidekiq-0.22.0/lib/opentelemetry/instrumentation/sidekiq/patches/processor.rb:21:in `process_one'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/processor.rb:72:in `run'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/component.rb:8:in `watchdog'
	/app/vendor/bundle/ruby/3.1.0/gems/sidekiq-7.0.2/lib/sidekiq/component.rb:17:in `block in safe_thread'
```

**Comment by gmcabrita** [comment]
Thanks Mike!
